zeroex/order_validator.go

```diff
@@ -654,6 +685,108 @@ func (o *OrderValidator) isSupportedAssetData(assetData []byte) bool {
 	return true
 }
 
+// findChunkSize uses a limited-depth binary-search to find a chunk size that is less then the
+// max content length
+func (o *OrderValidator) findChunkSize(signedOrders []*SignedOrder) int64 {
```

Comment:
Unless I'm mistaken, it looks like it is possible that the binary search will never reach `hi` with the way it is currently implemented, even in cases where all the given `signedOrders` could be included in the payload.

For example, consider a case where we have 500 orders and all of them could theoretically be included in the payload without hitting the size limit. The for loop will run 7 times, starting with `lo = 0`, `hi = 500`. On each iteration it will increase `mid`.

0. (0+500) / 2 = 250
1. (250+500) / 2 = 375
2. (375+500) / 2 = 428
3. (428+500) / 2 = 464
4. (464+500) / 2 = 482
5. (482+500) / 2 = 491
6. (491+500) / 2 = 495
7. (495+500) / 2 = 497

In fact, due to rounding, it is actually not possible to hit `max` even with infinite iterations in this case. I would expect that cases where all the given orders could fit within size constraints are fairly common. Maybe we can adjust the algorithm to check if `hi` fits within the size constraints directly?